validation/json.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validation
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// DuplicateKeyError is returned when duplicate key names are detected
+// inside a JSON object
+type DuplicateKeyError struct {
+	path []string
+	key  string
+}
+
+func (e *DuplicateKeyError) Error() string {
+	return fmt.Sprintf(`duplicate key "%s"`, strings.Join(append(e.path, e.key), "."))
+}
+
+// ValidateNoDuplicateKeys verifies the provided JSON object contains
+// no duplicated keys
+//
+// The function expects a single JSON object, and will error prior to
+// checking for duplicate keys should an invalid input be provided.
+func ValidateNoDuplicateKeys(s string) error {
```

Comment:
Two suggestions for the function name:
1. When used elsewhere, it will be in the `validation` package, so `Validate` is redundant
2. The function name should suggest that it is for JSON values

Use something like
```suggestion
func JSONNoDuplicateKeys(s string) error {
```

Reply:
Done!